pkg/shared/families/malware/yara/yara.go

```diff
@@ -79,18 +78,21 @@ func (s *Scanner) Run(sourceType utils.SourceType, userInput string) error {
 		// Execute the yara command
 		// nolint:gosec
 		yaraCommand := exec.Command(s.config.BinaryPath, args...)
-		out, err := sharedutils.RunCommand(yaraCommand)
-		if err != nil {
-			s.sendResults(retResults, fmt.Errorf("failed to run yara command: %s", err.Error()))
-			return
-		}
-
-		detectedMalware := util.ParseYaraScanOutput(string(out), s.logger)
-		retResults.Malware = detectedMalware
-		// In the case of Yara there is no scan summary.
-		retResults.Summary = &common.ScanSummary{}
-
-		s.sendResults(retResults, nil)
+		resChan := make(chan any, resultBufferSize)
+		ctx, cancel := context.WithCancel(context.Background())
+		go func() {
+			err := sharedutils.RunCommandAndParseOutputLineByLine(yaraCommand, util.ParseYaraScanOutput, resChan, s.logger)
```

Comment:
I think we should switch to using context to pass around the logger. For example:

```
ctx := context.TODO()
ctx = log.SetLoggerForContext(ctx, s.logger)
```

then in the code that needs the logger, as long as they have the context they can simply do:

```
logger := GetLoggerFromContextOrDiscard(ctx)
```